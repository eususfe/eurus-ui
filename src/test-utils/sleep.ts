/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
