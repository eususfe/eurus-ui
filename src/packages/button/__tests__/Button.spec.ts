import { mount } from '@vue/test-utils';
import EButton from '..';

describe('button', () => {

  it('button snapshot', () => {
    const wrapper = mount(EButton);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });
  it('should be render button props type', () => {

    const propsTypes = ['default', 'text', 'primary', 'info', 'success', 'warning', 'error', 'purple'] as const;

    propsTypes.forEach((type: string) => {
      const wrapper = mount(EButton, {
        props: {
          type
        }
      });
      expect(wrapper.classes()).toContain('eu-button--' + type);
      wrapper.unmount();

    });
  });

  it('should be render button size', () => {
    const sizeList = ['xs', 'sm', 'md', 'lg', 'xl'] as const;

    sizeList.forEach((size: string) => {
      const wrapper = mount(EButton, {
        props: {
          size,
        }
      });
      expect(wrapper.classes()).toContain('eu-button--' + size);
      wrapper.unmount();
    });
  });

  it('should button disabled', async () => {

    const onClick = vitest.fn();
    const wrapper = mount(EButton, {
      props: {
        disabled: true,
        onClick,
      }
    });

    expect(wrapper.classes()).toContain('disabled');
    await wrapper.trigger('click');
    expect(onClick).not.toHaveBeenCalled();
    wrapper.unmount();
  }
  );
  it('should button loading', async () => {
    const onClick = vitest.fn();
    const wrapper = mount(EButton, {
      props: {
        loading: true
      }
    });

    expect(wrapper.classes()).toContain('disabled');
    await wrapper.trigger('click');
    expect(onClick).not.toHaveBeenCalled();
    wrapper.unmount();
  }
  );
  it('should be render button native props', () => {
    const wrapper = mount(EButton, {
      props: {
        native: {
          type: 'submit'
        }
      }
    });

    expect(wrapper.attributes('type')).toContain('submit');

    wrapper.unmount();
  });

});
