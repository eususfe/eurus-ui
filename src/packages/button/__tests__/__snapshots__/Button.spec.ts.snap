// Vitest Snapshot v1

exports[`button > button snapshot 1`] = `
"<a class=\\"eu-button eu-button--default bg-default text-black eu-button--md\\">
  <!---->
  <!---->
</a>"
`;
